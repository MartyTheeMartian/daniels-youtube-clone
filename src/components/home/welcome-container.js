import {connect} from 'react-redux';
import Welcome from './welcome';

export const mapStateToProps = state => ({
  videoList: state.searchResults
});

export default connect(mapStateToProps)(Welcome);
