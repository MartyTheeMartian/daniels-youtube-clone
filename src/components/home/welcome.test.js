import React from 'react';
import Welcome from './welcome';
import { mapStateToProps } from './welcome-container'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('Welcome Component', () => {
  it('Renders without any videos', () => {
    const component = shallow(<Welcome videoList={[]} />)
    expect(component).toMatchSnapshot()
  })
  it('Renders with some videos', () => {
    const component = shallow(<Welcome videoList={[1]} />)
    expect(component).toMatchSnapshot()
  })
  it('mapStateToProps returns the correct state', () => {
    let state = {searchResults: [1,2,3]}
    const result = mapStateToProps(state)
    const expected = {videoList: [1,2,3]}
    expect(result).toEqual(expected)
  })
})
