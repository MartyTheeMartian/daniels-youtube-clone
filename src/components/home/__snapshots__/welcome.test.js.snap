// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Component Renders with some videos 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h2>
        Welcome to Daniel's YouTube Clone!
    </h2>
</div>,
  "nodes": Array [
    <div>
      <h2>
            Welcome to Daniel's YouTube Clone!
      </h2>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Welcome
        videoList={
                Array [
                        1,
                      ]
        }
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "videoList": Array [
            1,
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h2>
                    Welcome to Daniel's YouTube Clone!
          </h2>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div>
          <h2>
                    Welcome to Daniel's YouTube Clone!
          </h2>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Welcome
    videoList={
        Array [
            1,
          ]
    }
/>,
}
`;

exports[`Welcome Component Renders without any videos 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h2>
        Welcome to Daniel's YouTube Clone!
    </h2>
    <p>
        To get started please search for videos using the search bar in the upper righthand corner
    </p>
</div>,
  "nodes": Array [
    <div>
      <h2>
            Welcome to Daniel's YouTube Clone!
      </h2>
      <p>
            To get started please search for videos using the search bar in the upper righthand corner
      </p>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Welcome
        videoList={Array []}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "videoList": Array [],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h2>
                    Welcome to Daniel's YouTube Clone!
          </h2>
          <p>
                    To get started please search for videos using the search bar in the upper righthand corner
          </p>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <h2>
                    Welcome to Daniel's YouTube Clone!
          </h2>
          <p>
                    To get started please search for videos using the search bar in the upper righthand corner
          </p>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Welcome
    videoList={Array []}
/>,
}
`;
